### PR TITLE
Add: authentication

### PR DESCRIPTION
**Pull request by gauravsinhaweb**
Files to check:
- login.jsx
- signup.jsx
- auth-context.js
- navbar.jsx
- auth-reducer.jsx

### CODE REVIEW

File: src/context/auth-context.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+export const AuthProvider = ({ children }) => {
+  const [authState, authDispatch] = useReducer(authReducer, {
+    isUserLoggedIn: false,
+  });
+
+  const value = { authState, authDispatch };
+  return <AuthContext.Provider value={value}>{children}</AuthContext.Provider>;
+};
+
+export const useAuthContext = () => useContext(AuthContext);
```
Comment (JuvitaSaini09) [inline src/context/auth-context.js]: suggestion: you can export all at once which will look more cleaner.

**Review by JuvitaSaini09 — APPROVED** [review]

File: src/pages/auth/Login.jsx [hunk 2]
```diff
@@ -0,0 +1,126 @@
+import axios from "axios";
+import React, { useState } from "react";
+import { Link, useNavigate } from "react-router-dom";
+import { useAuthContext } from "../../context/useContext-index";
+
+export const Login = () => {
+  const navigate = useNavigate();
+  const { authDispatch } = useAuthContext();
+  const [loginData, setLoginData] = useState({
+    email: "",
```
Comment (yaswanthmyneni) [inline src/pages/auth/Login.jsx:10]: suggestion: send the useState to auth-context

File: src/pages/auth/SignUp.jsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
+import React from "react";
+import { Link } from "react-router-dom";
+import axios from "axios";
+
+export const SignUp = () => {
+  // axios.defaults.headers.common["authorization"] = encodedToken;
```
Comment (yaswanthmyneni) [inline src/pages/auth/SignUp.jsx]: changes: remove unused code

**Review by yaswanthmyneni — APPROVED** [review]
code looks good, you can merge after changes.

File: src/pages/auth/Login.jsx (hunk 2 above)
Comment (gauravsinhaweb) [reply src/pages/auth/Login.jsx:10]: the login form will not be used anywhere else, so I don't think it requires passing in context!

File: src/pages/auth/SignUp.jsx (hunk 3 above)
Comment (gauravsinhaweb) [reply src/pages/auth/SignUp.jsx]: Done

File: src/context/auth-context.js (hunk 1 above)
Comment (gauravsinhaweb) [reply src/context/auth-context.js]: Done!

**Pull request merged**